core/src/main/java/io/grpc/internal/MessageFramer.java

```diff
@@ -225,32 +241,65 @@ private int writeKnownLengthUncompressed(InputStream message, int messageLength)
     if (buffer == null) {
       buffer = bufferAllocator.allocate(header.position() + messageLength);
     }
-    writeRaw(headerScratch, 0, header.position());
-    return writeToOutputStream(message, outputStreamAdapter);
+    OutputStream outputStreamToWrite;
+    if (streamCompression) {
+      if (gzipOutputStream == null) {
+        try {
+          gzipOutputStream =
+              GZIPOutputStream.class
+                  .getConstructor(OutputStream.class, boolean.class)
+                  .newInstance(gzipOutputStreamAdapter, true);
+        } catch (Throwable t) {
```

Comment:
I'd much prefer we caught more-precise exceptions, or at the least only Exception instead of Throwable.